Fix for load errors when expansions are disabled (issue #1090)

Dirk-Dieters opened the pull request:
<!-- place 'x' mark between square [] brackets to affirm: -->
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

Dirk-Dieters committed:
fixed survival guide dependancy on COP

Added require to survival_guide.lua to make them independant of Chains of Promathia.

Dirk-Dieters committed:
Revert "fixed survival guide dependancy on COP"

This reverts commit f87df25fb6a6a019cd70ca6ea86fcebdd36845b6.

Dirk-Dieters committed:
Update survival_guide.lua

Kreidos commented:
Welcome @Dirk-Dieters ! Thanks for PRing this fix.

If possible, I'd like if we please hold this PR as work-in-progress until the remaining errors/requires are remedied so that we can have a single complete fix. :) Would you be willing to add the necessary commits to your branch, and thus this PR, to facilitate that?

Dirk-Dieters commented:
Perfectly reasonable, I'll take care of that as soon as I can.

Dirk-Dieters committed:
Added checks for disabled expansions

Added checks to the following zones before they perform zonewide actions to disable them when the expansion they belong to is disabled in the settings.lua.

Promyvion
Grauberg (S)
Misareaux Coast
Garden of Ru'Hmet
Uleguerand Range

Dirk-Dieters commented:
All load errors are now fixed in my fork. Summary of changes:
Added require line to survival_guide.lua
Added checks for all failing load attempts for disabled expansion zones

Dirk-Dieters committed:
Delete .gitignore

Dirk-Dieters committed:
Delete settings.lua

Dirk-Dieters committed:
Revert "Delete settings.lua"

This reverts commit f291ae76aa9b62be979b89c9fab71b64a3e09a0b.

Dirk-Dieters committed:
Revert "Delete .gitignore"

This reverts commit f2d8a580ce56faeb28ec49e55c52675c24ed4a06.

Dirk-Dieters committed:
Update settings.lua

reverted settings.lua to defaults for public repo.

Dirk-Dieters committed:
Revert "Revert "Delete .gitignore""

This reverts commit 340c1faf07080ff8826955e88f13d8e4b73b9e8b.

tankfest commented on scripts/globals/promyvion.lua:
```diff
@@ -13,39 +14,45 @@ tpz.promyvion = tpz.promyvion or {}
-    end
+		for _, v in pairs(ID.mob.MEMORY_RECEPTACLES) do
+			m = math.max(m, v[1])
+		end
```
Lot o' tab indents being used in these changed files.  All tabs need to be replaced with 4 spaces.

Notepad++ has this as a settings option so that when you press tab, it gives you 4 spaces instead.

Dirk-Dieters replied:
My mistake, I'll fix that.

Kreidos commented:
I'm not sure selectively putting ifs around so many pieces of disparate code is the right approach to solving a problem like this; Yes, the startup errors went away but I feel there's unintended collateral damage here. For one, it still leaves stub functions which do nothing and that could be affecting other functions elsewhere which rely on them without anyone knowing until the script tries to run.

If there are certain script files which are **confirmed** to contain _no viable code_ when a certain expansion is disabled and can be safely ignored then perhaps they can either be removed from the startup, or a single `if` could be used to return immediately and not execute the rest of the file.

It also appears that some files are being inadvertently removed (.gitignore)

Dirk-Dieters committed:
Merge pull request #1 from project-topaz/release

9/10

Dirk-Dieters commented:
I agree, the kludge is strong. I'll keep testing and watching for any aberrant behavior.

cocosolos commented:
Instead of touching promyvion.lua, you should just check if COP is enabled in the promyvion Zone.lua files where they call the promyvion init function.

ibm2431 commented:
If an expansion isn't enabled, the zone shouldn't be loaded at all.

Dirk-Dieters committed:
Merge pull request #2 from project-topaz/release

2

Dirk-Dieters committed:
1

TeoTwawki commented on conf/login.conf:
```diff
@@ -0,0 +1,70 @@
+mysql_host:      127.0.0.1
+mysql_port:      3306
+mysql_login:     root
+mysql_password:  wolfkin
```
yer gonna wanna change your password now. also amend / rebase your commits to not include your confs

TeoTwawki commented:
> If an expansion isn't enabled, the zone shouldn't be loaded at all.

I almost agreed with this but it creates other problems I don't want to deal with. I need zones to exist unless I explicitly disable them in my zone_settings table.

Dirk-Dieters commented:
I'm going to start fresh and try again.
